opm/parser/eclipse/RawDeck/StarToken.hpp

```diff
@@ -36,15 +36,39 @@ namespace Opm {
             return boost::lexical_cast<T>(valueString);
         }
         catch (boost::bad_lexical_cast&) {
-            throw std::invalid_argument("Unable to parse string" + valueString + " to typeid: " + typeid(T).name());
+            throw std::invalid_argument("Unable to convert string '" + valueString + "' to typeid: " + typeid(T).name());
         }
     }
 
     template <>
+    inline double readValueToken<double>(const std::string& valueString) {
+        // Eclipse supports Fortran syntax for specifying exponents of floating point
+        // numbers ('D' and 'E', e.g., 1.234d5) while C++ only supports the 'e' (e.g.,
+        // 1.234e5). the quick fix is to replace 'D' by 'E' and 'd' by 'e' before trying
+        // to convert the string into a floating point value. This may not be the most
+        // performant thing to do, but it is probably fast enough.
+        std::string vs(valueString);
+        std::replace(vs.begin(), vs.end(), 'D', 'E');
+        std::replace(vs.begin(), vs.end(), 'd', 'e');
+
+        try {
+            return boost::lexical_cast<double>(vs);
```

Comment:
> OK - fair enough; the truth of the matter is of course that I have shockingly limited knowledge of where the parser spends it's time. 

for keywords with DATA items, I'm reasonably confident the following is the place to point the finger to: https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/RawDeck/RawKeyword.cpp#L78 . This is because for records with several hundreds of thousands of values (ZCORN?), `m_partialRecordString` can become several megabytes large. If you consider the fact that the data occasionally needs to be copied for `operator+=`, its actually surprisingly fast! 

I'm not sure how easily this can be speed up, though.